From 81a6fb9c403b32671a04d6a2085fd44f5773496f Mon Sep 17 00:00:00 2001
From: Daniel Hsu <Daniel-Hsu@quantatw.com>
Date: Fri, 24 Jul 2026 14:49:19 +0800
Subject: [PATCH] configurations: revise CX7 NIC configuration

Add MCTPI2CTarget entry to allow MCTPReactor to automatically
assign an EID to the CX7 NIC over I2C, removing the need for manual
EID configuration.

Revise the CX7 NIC temperature sensor name.

Remove the lower critical threshold
Add upper non-recoverable threshold

Tested on Yv5:
EID is correctly assigned, the version object is properly exposed.
Temperature sensor is exposed as NIC0_TEMP_C.
---
 configurations/ocp/cx7_ocp.json | 28 ++++++++++++++++++----------
 1 file changed, 18 insertions(+), 10 deletions(-)

diff --git a/configurations/ocp/cx7_ocp.json b/configurations/ocp/cx7_ocp.json
index 3f11519..b13dd70 100644
--- a/configurations/ocp/cx7_ocp.json
+++ b/configurations/ocp/cx7_ocp.json
@@ -3,37 +3,45 @@
         {
             "Address": "$address",
             "Bus": "$bus",
-            "Name": "OCP CX7 NIC FRU $bus",
+            "Name": "OCP CX7 NIC FRU $bus / 10",
             "Type": "EEPROM"
         },
+        {
+            "Address": "0x32",
+            "Bus": "$bus",
+            "Name": "NIC",
+            "Type": "MCTPI2CTarget"
+        },
         {
             "Address": "0x1f",
             "Bus": "$bus",
             "Labels": [
                 "temp2"
             ],
-            "Name": "NIC Temp $bus",
-            "Name1": "NIC Temp $bus",
+            "Name": "NIC$bus / 10 TEMP_C",
+            "Name1": "NIC$bus / 10 TEMP_C",
             "Thresholds": [
                 {
-                    "Direction": "less than",
+                    "Direction": "greater than",
+                    "Hysteresis": 2.1,
                     "Index": 2,
-                    "Name": "lower critical",
+                    "Name": "upper critical",
                     "Severity": 1,
-                    "Value": 5
+                    "Value": 105
                 },
                 {
                     "Direction": "greater than",
+                    "Hysteresis": 2.3,
                     "Index": 2,
-                    "Name": "upper critical",
-                    "Severity": 1,
-                    "Value": 105
+                    "Name": "upper non recoverable",
+                    "Severity": 4,
+                    "Value": 115
                 }
             ],
             "Type": "TMP421"
         }
     ],
-    "Name": "OCP CX7 NIC $bus",
+    "Name": "OCP CX7 NIC $bus / 10",
     "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': 'Nvidia ConnectX-7 OCP3.0'})",
     "Type": "Board",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
-- 
2.53.0-Meta

